src/app/pages/storage/volumes/volume-import-wizard/volume-import-wizard.component.ts

```diff
@@ -348,6 +350,32 @@ export class VolumeImportWizardComponent implements WizardConfiguration {
       dialogRef.componentInstance.submit();
       dialogRef.componentInstance.success.pipe(untilDestroyed(this)).subscribe(() => {
         dialogRef.close(false);
+
+        if (this.pool) {
+          this.modalService.closeSlideIn();
+          this.modalService.refreshTable();
+          this.ws.call('pool.dataset.query', [[['pool', '=', this.pool]]]).pipe(untilDestroyed(this)).subscribe((datasets) => {
+            for (const dataset of datasets) {
+              if (dataset.encrypted && dataset.locked) {
+                this.dialogService.confirm({
+                  title: helptext.unlock_dataset_dialog_title,
+                  message: helptext.unlock_dataset_dialog_message,
+                  hideCheckBox: true,
+                  buttonMsg: helptext.unlock_dataset_dialog_button,
+                }).pipe(untilDestroyed(this)).subscribe((unlock) => {
+                  if (unlock) {
```

Comment:
Would be good to rewrite this new code in a way that would prefer returning early instead of going deeper when possible.